devdoc/waiter_queue_requirements.md

```diff
@@ -0,0 +1,69 @@
+# `waiter_queue` requirements
+
+## Overview
+
+`waiter_queue` is a module that allows a consumers to park calls and for producers to execute them when data becomes available. It has two parts: `waiter_queue_ll` that contains the logic for parking and executing calls and `waiter_queue_hl` that contains logic for making `waiter_queue` thread-safe.
```

Comment:
>t allows a consumers to park calls and for producers [](http://example.com/codeflow?start=30&length=52)

I think we are referencing in this statement producers/consumers (which relate to data/bytes) and at the same time we mention execution of calls.

I think if we mention both we'll get to a confusing concept. Let's leave the data out and focus on pausing/parking execution until a signal is received.

Reply:
Changed wording to remove data.